Source Code/RoleAssignmentPatch.cs

```diff
@@ -241,10 +243,20 @@ public static void Postfix([HarmonyArgument(0)]Il2CppReferenceArray<GameData.Pla
         }
 
         private static void setRoleToRandomPlayer(byte roleId, List<PlayerControl> playerList) {
+            
+            if (CustomOptionHolder.blockedRolePairings.ContainsKey(roleId)) {
+                foreach(var blockedRoleId in CustomOptionHolder.blockedRolePairings[roleId]) {
+                    if (AssignedRoles.Contains((byte)blockedRoleId) && roleId.Equals(roleId)) {
+                        return;
```

Comment:
Since this is only supposed to assign and doesnt return success state, the return without assign will cause less roles to be distributed, as the assignment system assumes this to have assigned a role and will reduce the number of roles to assign, accordingly.